src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,426 @@
+                {
+                    foreach (var peFile in Directory.GetFiles(assemDir, searchPattern, SearchOption.AllDirectories))
+                    {
+                        if (!peFile.Contains(".ni.dll"))
+                        {
+                            files.Add(peFile);
+                        }
+                    }
+                }
+            }
+            return files;
+        }
+
+        private static void GenXmlFromPdb(string assemblyPath, string pdbPath, string xmlPath)
+        {
+            using var peStream = new FileStream(assemblyPath, FileMode.Open, FileAccess.Read);
+            using var pdbStream = new FileStream(pdbPath, FileMode.Open, FileAccess.Read);
+            using var dstFileStream = new FileStream(xmlPath, FileMode.Create, FileAccess.ReadWrite);
+            using var sw = new StreamWriter(dstFileStream, Encoding.UTF8);
+            PdbToXmlOptions options = PdbToXmlOptions.ResolveTokens | PdbToXmlOptions.IncludeTokens;
+
+            PdbToXmlConverter.ToXml(sw, pdbStream, peStream, options);
+        }
+
+        private static void RemoveTempDirectory()
+        {
+            if (Directory.Exists(tempDirectory))
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+
+        private static void Symbolicator(IConsole console, List<string> xmlList, string inputPath, string outputPath)
+        {
+            if (xmlList.Count == 0)
+            {
+                RemoveTempDirectory();
+                throw new FileNotFoundException("Xml file not found\n");
+            }
+
+            GetLineFromStack(console, xmlList, inputPath, outputPath);
+
+            RemoveTempDirectory();
+        }
+
+        internal sealed class StackTraceInfo
+        {
+            public string Type;
+            public string Method;
+            public string Param;
+            public string Assembly;
+            public string Token;
+            public string Offset;
+            public string Document;
+            public string Filepath;
+            public string Filename;
+            public string StartLine;
+            public string EndLine;
+        }
+
+        private static void GetLineFromStack(IConsole console, List<string> xmlList, string inputPath, string outputPath)
+        {
+            try
+            {
+                StreamWriter fswi = null;
+                if (!inputPath.EndsWith(".symbolicated"))
+                {
+                    fswi = new StreamWriter(new FileStream(inputPath + ".symbolicated", FileMode.Create, FileAccess.Write));
+                }
+
+                string output = string.Empty;
+                StreamWriter fswo = null;
+                if (outputPath != null)
+                {
+                    fswo = new StreamWriter(new FileStream(outputPath, FileMode.Create, FileAccess.Write));
+                    output = $"\nOutput: {outputPath}\n";
+                }
```

Comment:
This looks like it always writes the fixed up text to a `.symbolicated` file next to the input file. and then optionally also writes to an output file.

The tool should do one _or_ the other, but not both since that would mean a user generates two files when they specify an output path.